src/java.base/share/classes/java/lang/reflect/code/op/ExtendedOps.java

```diff
@@ -2791,6 +2791,53 @@ public Value processor() {
         public List<Value> fragments() {
             return operands().subList(1, operands().size());
         }
+
+        public List<Body> expressions() {
+            return expressions;
+        }
+
+        @Override
+        public Block.Builder lower(Block.Builder b, OpTransformer opT) {
+
+            Block.Builder[] builders = new Block.Builder[1];
+            builders[0] = b;
+            List<Value> expressions = new ArrayList<>();
+            for (Body expression : expressions()) {
+                builders[0].transformBody(expression, List.of(), opT.andThen((block, op) -> {
+                    if (op instanceof YieldOp yop) {
+                        expressions.add(block.context().getValue(yop.yieldValue()));
+                        builders[0] = block;
+                    } else if (op instanceof Lowerable lop) {
+                        block = lop.lower(block, opT);
+                    } else {
+                        block.op(op);
+                    }
+                    return block;
+                }));
+            }
+
+            Block.Builder builder = builders[0];
+
+            Op.Result fragmentsList = builder.op(invoke(
```

Comment:
The last parameter type of the method is `Object[]` but you are not packaging up the expression values into an array, which will require boxing of primitive values. This likely just happily works out in the interpreter because how it uses method handles, but likely fails when generating bytecode.

I am unsure if we should support this varargs-like behaviour as part of the invoke operation (and boxing). I am leaning towards not supporting.

My recommendation is to package up the expression values. Here's some example code (used for creating code models that build models). You would need to add boxing support.

```
    Value buildList(JavaType elementType, List<Value> elements) {
        JavaType listType = type(J_U_LIST, elementType);
        if (elements.size() < 11) {
            MethodRef listOf = MethodRef.method(J_U_LIST, "of",
                    J_U_LIST, Collections.nCopies(elements.size(), J_L_OBJECT));
            return builder.op(invoke(listType, listOf, elements));
        } else {
            Value array = buildArray(elementType, elements);
            return builder.op(invoke(listType, LIST_OF_ARRAY, array));
        }
    }


    Value buildArray(JavaType elementType, List<Value> elements) {
        Value array = builder.op(newArray(elementType,
                builder.op(constant(INT, elements.size()))));
        for (int i = 0; i < elements.size(); i++) {
            builder.op(arrayStoreOp(array, elements.get(i),
                    builder.op(constant(INT, i))));
        }
        return array;
    }
```